src/client_state/execution.rs

```diff
@@ -0,0 +1,186 @@
+use ibc_core::primitives::proto::Any;
+use ibc_core_host::types::path::{ClientConsensusStatePath, ClientStatePath};
+
+use crate::client_message::Header;
+use crate::client_state::ClientState;
+
+impl<E> ClientStateExecution<E> for ClientState
+where
+    E: ExtClientExecutionContext,
+    E::ClientStateRef: From<ClientState>,
+    E::ClientStateMut: From<ibc_client_tendermint::types::ClientState>,
+    TendermintConsensusStateType: Convertible<E::ConsensusStateRef>,
+    ClientError: From<<TendermintConsensusStateType as TryFrom<E::ConsensusStateRef>>::Error>,
+{
+    fn initialise(
+        &self,
+        ctx: &mut E,
+        client_id: &ClientId,
+        consensus_state: Any,
+    ) -> Result<(), ClientError> {
+        initialise(self, ctx, client_id, consensus_state)
+    }
+
+    fn update_state(
+        &self,
+        ctx: &mut E,
+        client_id: &ClientId,
+        header: Any,
+    ) -> Result<Vec<Height>, ClientError> {
+        update_state(self, ctx, client_id, header)
+    }
+
+    fn update_state_on_misbehaviour(
+        &self,
+        ctx: &mut E,
+        client_id: &ClientId,
+        client_message: Any,
+    ) -> Result<(), ClientError> {
+        update_on_misbehaviour(self, ctx, client_id, client_message)
+    }
+
+    // Commit the new client state and consensus state to the store
+    fn update_state_on_upgrade(
+        &self,
+        _ctx: &mut E,
+        _client_id: &ClientId,
+        _upgraded_client_state: Any,
+        _upgraded_consensus_state: Any,
+    ) -> Result<Height, ClientError> {
+        unimplemented!()
+    }
+
+    fn update_on_recovery(
+        &self,
+        _ctx: &mut E,
+        _subject_client_id: &ClientId,
+        _substitute_client_state: Any,
+        _substitute_consensus_state: Any,
+    ) -> Result<(), ClientError> {
+        unimplemented!()
+    }
+}
+
+/// Seed the host store with initial client and consensus states.
+///
+/// Note that this function is typically implemented as part of the
+/// [`ClientStateExecution`] trait, but has been made a standalone function
+/// in order to make the ClientState APIs more flexible.
+pub fn initialise<E>(
+    client_state: &ClientState,
+    ctx: &mut E,
+    client_id: &ClientId,
+    consensus_state: Any,
+) -> Result<(), ClientError>
+where
+    E: ExtClientExecutionContext,
+    E::ClientStateRef: From<ClientState>,
+    TendermintConsensusStateType: Convertible<E::ConsensusStateRef>,
+    ClientError: From<<TendermintConsensusStateType as TryFrom<E::ConsensusStateRef>>::Error>,
+{
+    let host_timestamp = ExtClientValidationContext::host_timestamp(ctx)?;
+    let host_height = ExtClientValidationContext::host_height(ctx)?;
+
+    let tendermint_consensus_state: TendermintConsensusStateType = consensus_state
+        .clone()
+        .try_into()
+        .map_err(|_: ClientError| ClientError::UnknownConsensusStateType {
+            consensus_state_type: consensus_state.type_url,
+        })?;
```

Comment:
@rnbguy Would you happen to know if this can be simplified? Doing just `consensus_state.try_into()` was giving me errors about the `ClientError`, and this is what I came up with to work around them...

Reply:
Very interesting find. Looks like the Rust compiler is getting confused between the implementations for `TryFrom<Any>` and `TryFrom<E::ConsensusStateRef>`. 

For this particular code, you can remove [this](https://github.com/cosmos/rollkit-ibc/pull/22/files/45b9897e1dabe37361f77869eaa11a7dc2c50d75#diff-d773d4d41580ba906c012458fe64a4ef7f9fc874fc594d8e6d3b1774c1475572R87) trait bound, and that allows `.try_into()?`.

If we really needed that trait bound, you could have done:

```rs
let tendermint_consensus_state =
  <TendermintConsensusStateType as TryFrom<Any>>::try_from(consensus_state.clone())?;
```

Reply:
I pushed a commit for this 31babdb

Reply:
Thanks for the suggestion, @rnbguy! I will try the removal of the trait bound for now. :)